service/controller/v15/cluster_resource_set.go

```diff
@@ -132,30 +132,33 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 			Address: config.VaultAddress,
 		}
 
-		encrypterObject, err = vault.NewEncrypter(c)
+		e, err := vault.NewEncrypter(c)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
-		encrypterRoleManager = encrypterObject.(encrypter.RoleManager)
+		encrypterObject = e
+		encrypterResource = e
```

Comment:
I get that we want to split the object and resource. As well as eventually removing the resource.

Still this looks weird to me and I'd appreciate some explanatory comments.